packages/ics721/src/utils.rs

```diff
@@ -1,52 +1,72 @@
+use cw721::msg::NumTokensResponse;
 use cw_ownable::Ownership;
 
 use crate::state::{CollectionData, UniversalCollectionInfoResponse};
 
 pub fn get_collection_data(deps: &DepsMut, collection: &Addr) -> StdResult<CollectionData> {
-    // cw721 v0.17 and higher holds ownership in the contract
-    let ownership: StdResult<Ownership<Addr>> = deps
-        .querier
-        .query_wasm_smart(collection, &cw721_base::msg::QueryMsg::Ownership::<Addr> {});
-    let owner = match ownership {
+    // cw721 v0.19 and higher holds creator ownership (cw-ownable storage) in the contract
+    let ownership_result: StdResult<Ownership<Addr>> = deps.querier.query_wasm_smart(
+        collection,
+        &cw721_metadata_onchain::msg::QueryMsg::GetCreatorOwnership {},
+    );
+    let owner = match ownership_result {
         Ok(ownership) => ownership.owner.map(|a| a.to_string()),
         Err(_) => {
-            // cw721 v0.16 and lower holds minter
-            let minter_response: cw721_base_016::msg::MinterResponse = deps
-                .querier
-                .query_wasm_smart(collection, &cw721_base_016::QueryMsg::Minter::<Empty> {})?;
-            deps.api.addr_validate(&minter_response.minter)?;
-            Some(minter_response.minter)
+            // cw721 v0.17 and v0.18 holds minter ownership (cw-ownable storage) in the contract
+            let ownership: StdResult<Ownership<Addr>> = deps.querier.query_wasm_smart(
+                collection,
+                &cw721_base_018::msg::QueryMsg::Ownership::<Addr> {}, // nb: could also use `GetMinterOwnership`, but some custom contracts may only know about `Ownership`
+            );
+            match ownership {
+                Ok(ownership) => ownership.owner.map(|a| a.to_string()),
+                Err(_) => {
+                    // cw721 v0.16 and lower holds minter (simple string storage)
+                    let minter_response: cw721_base_016::msg::MinterResponse =
+                        deps.querier.query_wasm_smart(
+                            collection,
+                            &cw721_base_016::QueryMsg::Minter::<Empty> {},
+                        )?;
+                    deps.api.addr_validate(&minter_response.minter)?;
+                    Some(minter_response.minter)
+                }
+            }
         }
     };
     let contract_info = deps.querier.query_wasm_contract_info(collection)?;
-    let UniversalCollectionInfoResponse { name, symbol } = deps.querier.query_wasm_smart(
+    let UniversalCollectionInfoResponse {
+        name,
+        symbol,
+        extension,
+        updated_at: _,
+    } = deps.querier.query_wasm_smart(
         collection,
-        &cw721_base::msg::QueryMsg::<Empty>::ContractInfo {},
+        #[allow(deprecated)]
+        // For now we use `ContractInfo` which is known across all version, whilst `GetCollectionInfoAndExtension` is only available in v0.19 and higher
+        &cw721_metadata_onchain::msg::QueryMsg::ContractInfo {},
```

Comment:
This is where collection metadata is retrieved. cw721 v18 and below only holds name and symbol. Whilst v19 holds creator and royalty info.